catboost: adding option to build R library

nviets opened the pull request:
Adding option to build to build the R library. See included comments for an example of usage.

@natsukium, @jbedo 

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

alyssais committed:
nixos/qemu-vm: fix infinite recursion

The virtualisation.directBoot.initrd option was added for netboot
images, but the assertion to check directBoot enabled if it was used
caused an infinite recursion if it was.  Minimal reproduction:

	import nixos/tests/make-test-python.nix ({ pkgs, ... }: {
	  name = "";

	  nodes = {
	    machine = { config, ...}: {
	      imports = [ nixos/modules/installer/netboot/netboot-minimal.nix ];

	      virtualisation.directBoot = {
	        enable = true;
	        initrd = "${config.system.build.netbootRamdisk}/${config.system.boot.loader.initrdFile}";
	      };
	    };
	  };
	  testScript = "";
	}) {}

The fix is to swap the two conditions, so that cfg.directBoot.enable
is checked first, and the initrd comparision will be short circuited.

This wasn't noticed during review because in earlier versions of the
virtualisation.directBoot patch, the assertion was accidentally in the
conditional above, so wasn't evaluated unless port forwarding was in
use.

Commit:
Adding option to build to build the R library. See included comments for
an example of usage.

natsukium commented on pkgs/development/libraries/catboost/default.nix:
```diff
@@ -79,24 +97,45 @@ stdenv.mkDerivation (finalAttrs: {
+    "-DCATBOOST_COMPONENTS=app;libs${lib.optionalString pythonSupport ";python-package"}${lib.optionalString rLibrary ";R-package"}"
   ] ++ lib.optionals cudaSupport [
-    "-DHAVE_CUDA=on"
+    "-DHAVE_CUDA=yes"
```
I don't mind changing it, but I want to keep it consistent.

nviets replied:
[Catboost's latest documentation](https://catboost.ai/en/docs/installation/build-native-artifacts) indicates yes|no. Is there any reason to go with on|off instead?

natsukium replied:
There is no particular reason for on/off, just personal preference. (I really wanted to use uppercase, but for some reason, it's lowercase.)

natsukium commented on pkgs/development/libraries/catboost/default.nix:
```diff
@@ -79,24 +97,45 @@ stdenv.mkDerivation (finalAttrs: {
   installPhase = ''
     runHook preInstall
-
+  '' + lib.optionalString (!rLibrary) ''
     mkdir $dev
     cp -r catboost $dev
     install -Dm555 catboost/app/catboost -t $out/bin
```
```suggestion
    mkdir $dev
    cp -r catboost $dev
  '' + lib.optionalString (!rLibrary) ''
    install -Dm555 catboost/app/catboost -t $out/bin
```
I guess the python library is not built with `rLibrary=true; pythonSupport=true;`.

I think the installPhase will need to be reconstructed later.

nviets commented:
Thanks for your suggestions, @natsukium. I have updated the PR.

nicoonoclaste committed:
nixos/vagrant-guest: Set `security.sudo-rs.wheelNeedsPassword` too

paveloom committed:
qbittorrent: 4.5.5 -> 4.6.0 + other changes

Other changes:
- Refactor
- Switch to CMake
- Build with Qt6 by default and add a separate derivation for Qt5

paveloom committed:
libtorrent-rasterbar: fix the path to the library in the CMake module

nviets commented:
@natsukium - gentle reminder. thanks!

natsukium reviewed (changes requested):
Please change the commit message to `catboost: adding option to build R library`.

natsukium commented:
It works fine on my x86_64-linux with/without cuda & aarch64-darwin machines with the following command.
```
nix-shell -p 'rWrapper.override { packages = [ (catboost.override { rLibrary = true; }) ]; }'
```

vbgl committed:
ocamlPackages.sodium: disable for OCaml ≥ 5.0

vbgl committed:
ocamlPackages.camlimages: fix build with OCaml ≥ 5.0

vbgl committed:
ocamlPackages.cry: fix build with OCaml ≥ 5.0

vbgl committed:
ocamlPackages.piqi: disable for OCaml ≥ 5.0

alyssais committed:
nixos/stage-1: create initramfs /lib at build time

Modules built in to the kernel can attempt to load firmware before
init is started.  To guarantee the firmware is accessible to them
where they expect, /lib has to exist in the initramfs — it can't be
created later by init, because by that point the module may already
have tried and given up.

lblasc committed:
lua-rtoml: init 0.2

vbgl committed:
ocamlPackages.dum: fix build with OCaml ≥ 5.0

Munksgaard committed:
smlfmt: 1.0.0 -> 1.1.0

AndersonTorres committed:
xosview: migrate to by-name

AndersonTorres committed:
xosview2: migrate to by-name

AndersonTorres committed:
xosview: set meta.mainProgram

AndersonTorres committed:
xosview: split man output

AndersonTorres committed:
xosview2: set meta.mainProgram

AndersonTorres committed:
xosview2: 2.3.2 -> 2.3.3

Mic92 committed:
nixos/mediawiki: fix rewrites for static ressources and rest API

chewblacka committed:
clamtk: init at 6.16

Mic92 committed:
nixos/mediawiki: pin php to 8.1

ipetkov committed:
syncoid: disable PrivateUsers in systemd unit

* leaving PrivateUsers=true appears to break when using zfs-2.2.0

r-ryantm committed:
python310Packages.dask-histogram: 2023.6.0 -> 2023.10.0

al3xtjames committed:
python311Packages.img2pdf: fix evaluation on darwin

The img2pdf derivation uses the sRGB ICC profile provided by colord
starting with #264119. colord and several of its dependencies only
support Linux, so this broke evaluation on Darwin. This change drops the
use of colord on Darwin to fix the derivation; the system-provided sRGB
profile is used instead.

adisbladis committed:
Merge pull request #265585 from marsam/add-emacsPackages.lspce

emacsPackages.lspce: init at unstable-2023-10-30

jbedo committed:
Merge pull request #261763 from nviets/nngUpdate

nng: 1.6.0-prerelease update for rPackages.nanonext

Ma27 committed:
Merge pull request #265708 from Stunkymonkey/nixos-test-prometheus-exporters

nixos/prometheus-exporters: fix knot & pgbounce tests

Ma27 committed:
Merge pull request #265071 from 999eagle/feat/matrix-sliding-sync-update

nixos/matrix-sliding-sync: add dependency on matrix-synapse if running locally

uri-canva committed:
Merge pull request #265130 from r-ryantm/auto-update/opentelemetry-collector

opentelemetry-collector: 0.87.0 -> 0.88.0

r-ryantm committed:
python310Packages.djangorestframework-dataclasses: 1.3.0 -> 1.3.1

aanderse committed:
Merge pull request #265705 from r-ryantm/auto-update/kodiPackages.pvr-hts

kodiPackages.pvr-hts: 20.6.3 -> 20.6.4

markuskowa committed:
Merge pull request #265649 from markuskowa/upd-nwchem

nwchem: 7.2.1 -> 7.2.2

NickCao committed:
Merge pull request #265682 from r-ryantm/auto-update/tbox

tbox: 1.7.4 -> 1.7.5

NickCao committed:
Merge pull request #265681 from r-ryantm/auto-update/sqlite-utils

sqlite-utils: 3.35.1 -> 3.35.2

NickCao committed:
Merge pull request #265670 from r-ryantm/auto-update/python311Packages.python-on-whales

python311Packages.python-on-whales: 0.65.0 -> 0.66.0

natsukium committed:
Merge pull request #265575 from natsukium/zxing-cpp

python311Packages.zxing-cpp: rename from zxing_cpp

jfvillablanca committed:
llm-ls: init at 0.4.0

r-ryantm committed:
python310Packages.duckduckgo-search: 3.8.5 -> 3.9.4

r-ryantm committed:
python310Packages.dvclive: 3.1.0 -> 3.2.0

Artturin committed:
Merge pull request #265570 from amjoseph-nixpkgs/pyqtwebengine-take-different-python3

dotlambda committed:
Merge pull request #264418 from dotlambda/restic-0.16.2

restic: 0.16.1 -> 0.16.2

AndersonTorres committed:
sxhkd: refactor

- migrate to by-name
- split outputs
- add meta.mainProgram

r-ryantm committed:
python310Packages.easydict: 1.10 -> 1.11

dotlambda committed:
Merge pull request #265752 from al3xtjames/img2pdf-icc-path

python311Packages.img2pdf: fix evaluation on darwin

kashw2 committed:
terragrunt: 0.53.0 -> 0.53.2

r-ryantm committed:
python310Packages.ezyrb: 1.3.0.post2309 -> 1.3.0.post2311

samueldr committed:
Merge pull request #262684 from samueldr/feature/nixos-rebuild-completions

nixos-rebuild: Locally own the nixos-rebuild (bash) completions

r-ryantm committed:
python310Packages.faster-whisper: 0.8.0 -> 0.9.0

r-ryantm committed:
python310Packages.fasttext-predict: 0.9.2.1 -> 0.9.2.2

r-ryantm committed:
python310Packages.finvizfinance: 0.14.6 -> 0.14.7

marsam committed:
Merge pull request #260727 from bcdarwin/opencv-python-updates

python310Packages.opencv3: remove

fabianhjr committed:
Merge pull request #265720 from r-ryantm/auto-update/python310Packages.django-crispy-forms

python310Packages.django-crispy-forms: 2.0 -> 2.1

fabianhjr committed:
Merge pull request #265715 from lunik1/iosevka-27.3.4

iosevka: 27.3.2 -> 27.3.4

marsam committed:
Merge pull request #265758 from r-ryantm/auto-update/python310Packages.duckduckgo-search

python310Packages.duckduckgo-search: 3.8.5 -> 3.9.4

marsam committed:
Merge pull request #265749 from r-ryantm/auto-update/python310Packages.djangorestframework-dataclasses

python310Packages.djangorestframework-dataclasses: 1.3.0 -> 1.3.1

fabianhjr committed:
Merge pull request #265553 from r-ryantm/auto-update/ocserv

ocserv: 1.2.1 -> 1.2.2

fabianhjr committed:
Merge pull request #265539 from kilianar/portfolio-0.65.5

portfolio: 0.65.4 -> 0.65.5

marsam committed:
Merge pull request #265723 from r-ryantm/auto-update/python310Packages.django-ipware

python310Packages.django-ipware: 5.0.0 -> 5.0.2

marsam committed:
Merge pull request #265691 from r-ryantm/auto-update/wasmedge

wasmedge: 0.13.4 -> 0.13.5

marsam committed:
Merge pull request #265684 from notalltim/grpc-cross-build-cpp-plugin

grpc: explicitly use the build platform for `grpc_cpp_plugin`

marsam committed:
Merge pull request #265593 from r-ryantm/auto-update/carapace

carapace: 0.28.2 -> 0.28.3

marsam committed:
Merge pull request #265594 from r-ryantm/auto-update/gitoxide

gitoxide: 0.30.0 -> 0.31.1

marsam committed:
Merge pull request #265581 from r-ryantm/auto-update/python310Packages.cobs

python310Packages.cobs: 1.2.0 -> 1.2.1

r-ryantm committed:
python310Packages.flask-paginate: 2023.10.8 -> 2023.10.24

r-ryantm committed:
python310Packages.fschat: 0.2.30 -> 0.2.32

dotlambda committed:
Merge pull request #265498 from r-ryantm/auto-update/python310Packages.gdal

gdal: 3.7.2 -> 3.7.3

r-ryantm committed:
python310Packages.glcontext: 2.4.0 -> 2.5.0

honnip committed:
pretendard: 1.3.8 -> 1.3.9

r-ryantm committed:
python310Packages.gocardless-pro: 1.47.0 -> 1.48.0

K900 committed:
Merge pull request #265700 from K900/nix-prefetch-git-progress

nix-prefetch-git: download submodules with --progress

fabaff committed:
python311Packages.alexapy: 1.27.7 -> 1.27.8

Diff: https://gitlab.com/keatontaylor/alexapy/-/compare/refs/tags/v1.27.7...v1.27.8

Changelog: https://gitlab.com/keatontaylor/alexapy/-/blob/v1.27.8/CHANGELOG.md

fabaff committed:
python311Packages.boschshcpy: 0.2.73 -> 0.2.75

Diff: https://github.com/tschamm/boschshcpy/compare/0.2.73...0.2.75

fabaff committed:
python311Packages.amqp: 5.1.1 -> 5.2.0

fabaff committed:
python311Packages.dvclive: 3.1.0 -> 3.2.0

Diff: https://github.com/iterative/dvclive/compare/refs/tags/3.1.0...3.2.0

Changelog: https://github.com/iterative/dvclive/releases/tag/3.2.0

fabaff committed:
python311Packages.aiocomelit: 0.3.0 -> 0.3.1

Diff: https://github.com/chemelli74/aiocomelit/compare/refs/tags/v0.3.0...v0.3.1

Changelog: https://github.com/chemelli74/aiocomelit/blob/0.3.1/CHANGELOG.md

fabaff committed:
python311Packages.amqp: add changelog to meta

fabaff committed:
python311Packages.amqp: update disabled

Commit:
python311Packages.gphoto2: fix setuptools.__version__ build breakage

setuptools.__version__ = '68.2.2.post0'
setup.py fails with a parse error when trying to convert 'post0' to an
int. patch the code so it only will convert the first to values returned
by split('.')

fabaff committed:
python311Packages.blinkpy: 0.22.2 -> 0.22.3

Diff: https://github.com/fronzbot/blinkpy/compare/refs/tags/v0.22.2...v0.22.3

Changelog: https://github.com/fronzbot/blinkpy/blob/refs/tags/v0.22.3/CHANGES.rst

fabaff committed:
python311Packages.evohome-async: 0.4.3 -> 0.4.4

Diff: https://github.com/zxdavb/evohome-async/compare/refs/tags/0.4.3...0.4.4

fabaff committed:
python311Packages.meshtastic: 2.2.11 -> 2.2.12

Diff: https://github.com/meshtastic/Meshtastic-python/compare/refs/tags/2.2.11...2.2.12

Changelog: https://github.com/meshtastic/python/releases/tag/2.2.12

fabaff committed:
python311Packages.lsassy: 3.1.8 -> 3.1.9

Diff: https://github.com/Hackndo/lsassy/compare/refs/tags/v3.1.8...v3.1.9

Changelog: https://github.com/Hackndo/lsassy/releases/tag/v3.1.9

fabaff committed:
python311Packages.rns: 0.6.5 -> 0.6.6

Diff: https://github.com/markqvist/Reticulum/compare/refs/tags/0.6.5...0.6.6

Changelog: https://github.com/markqvist/Reticulum/releases/tag/0.6.6

fabaff committed:
python311Packages.publicsuffixlist: 0.10.0.20231104 -> 0.10.0.20231105

fabaff committed:
python311Packages.pycep-parser: 0.4.1 -> 0.4.2

Diff: https://github.com/gruebel/pycep/compare/refs/tags/0.4.1...0.4.2

Changelog: https://github.com/gruebel/pycep/blob/0.4.2/CHANGELOG.md

fabaff committed:
python311Packages.plugwise: 0.34.4 -> 0.34.5

Diff: https://github.com/plugwise/python-plugwise/compare/refs/tags/v0.34.4...v0.34.5

Changelog: https://github.com/plugwise/python-plugwise/releases/tag/v0.34.5

fabaff committed:
python311Packages.pyoverkiz: 1.12.1 -> 1.12.2

Diff: https://github.com/iMicknl/python-overkiz-api/compare/refs/tags/v1.12.1...v1.12.2

Changelog: https://github.com/iMicknl/python-overkiz-api/releases/tag/v1.12.2

fabaff committed:
python311Packages.pysml: 0.1.0 -> 0.1.1

Diff: https://github.com/mtdcr/pysml/compare/refs/tags/0.1.0...0.1.1

fabaff committed:
python311Packages.tplink-omada-client: 1.3.5 -> 1.3.6

Changelog: https://github.com/MarkGodwin/tplink-omada-api/releases/tag/release%2Fv1.3.6

fabaff committed:
python311Packages.tldextract: 5.0.1 -> 5.1.0

Diff: https://github.com/john-kurkowski/tldextract/compare/refs/tags/5.0.1...5.1.0

Changelog: https://github.com/john-kurkowski/tldextract/blob/5.1.0/CHANGELOG.md

aaronjheng committed:
pb: 0.1.0 -> 0.2.0

thiagokokada committed:
Merge pull request #244298 from oxalica/remove-unused

{birdtray,isgx,mycrypto,osu-lazer,stretchly,tree-sitter}: remove oxalica as maintainer

fabaff committed:
python311Packages.skodaconnect: 1.3.7 -> 1.3.8

Diff: skodaconnect/skodaconnect@refs/tags/1.3.7...1.3.8

Changelog: https://github.com/lendy007/skodaconnect/releases/tag/1.3.8

r-ryantm committed:
python310Packages.gptcache: 0.1.41 -> 0.1.42

natsukium committed:
Merge pull request #265296 from r-ryantm/auto-update/kubectl-cnpg

kubectl-cnpg: 1.20.2 -> 1.21.0

JakeHPark committed:
autokey: 0.95.10 -> 0.96.0

pbsds committed:
Merge pull request #263015 from surfaceflinger/lunarclient-autoupdate

lunar-client: migrate to by-name and add updateScript

r-ryantm committed:
python310Packages.greeneye-monitor: 5.0 -> 5.0.1

secufoe committed:
rshim-user-space: make bfb-install optional

elohmeier committed:
mkpasswd: fix build with clang

fabaff committed:
Merge pull request #265792 from fabaff/boschshcpy-bump

python311Packages.boschshcpy: 0.2.73 -> 0.2.75

fabaff committed:
Merge pull request #265793 from fabaff/amqp-bump

python311Packages.amqp: 5.1.1 -> 5.2.0

fabaff committed:
Merge pull request #265794 from fabaff/aiocomelit-bump

python311Packages.aiocomelit: 0.3.0 -> 0.3.1

fabaff committed:
Merge pull request #265795 from fabaff/dvclive-bump

python311Packages.dvclive: 3.1.0 -> 3.2.0

teto committed:
Merge pull request #263320 from jfvillablanca/llm-ls

llm-ls: init at 0.4.0

r-ryantm committed:
python310Packages.grpcio-channelz: 1.59.0 -> 1.59.2

rnhmjoj committed:
Merge pull request #263138 from tomfitzhenry/hostapd-optional-managementframeprotection

nixos/hostapd: remove managementFrameProtection in favour of clearer default

r-ryantm committed:
python310Packages.grpcio-reflection: 1.59.0 -> 1.59.2

Artturin committed:
Merge pull request #254264 from amesgen/bazecor

bazecor: init at 1.3.6

klchen0112 committed:
librime: set darwin as support platforms

Kranzes committed:
Merge pull request #264658 from adamcstephens/kanidm/1.1.0-rc.15

kanidm: 1.1.0-beta.13 -> 1.1.0-rc.15

GaetanLepage committed:
python311Packages.jaxopt: 0.8.1 -> 0.8.2

Changelog: https://github.com/google/jaxopt/releases/tag/jaxopt-v0.8.2

wegank committed:
inkscape: fix runtime error on darwin

r-ryantm committed:
python310Packages.httpx-socks: 0.7.8 -> 0.8.0

natsukium committed:
Merge pull request #264463 from GaetanLepage/objax

python311Packages.objax: disable tests to fix build

Mic92 committed:
rsync: fix regression with _FORTIFY_SOURCE=2

See https://github.com/WayneD/rsync/issues/511 for details.
Not using fetchpatch, because I didn't found a raw link for opensuse's repository

r-ryantm committed:
python310Packages.ignite: 0.4.12 -> 0.4.13

r-ryantm committed:
python310Packages.intellifire4py: 3.1.29 -> 3.1.30

wegank committed:
Merge pull request #245431 from leiserfg/staging

inkscape: 1.2.2 -> 1.3, lib2geom: 1.2.2 -> 1.3

NickCao committed:
sing-box: 1.6.0 -> 1.6.1

Diff: https://github.com/SagerNet/sing-box/compare/v1.6.0...v1.6.1

NickCao committed:
Merge pull request #265842 from r-ryantm/auto-update/python310Packages.httpx-socks

python310Packages.httpx-socks: 0.7.8 -> 0.8.0

NickCao committed:
Merge pull request #265823 from r-ryantm/auto-update/python310Packages.greeneye-monitor

python310Packages.greeneye-monitor: 5.0 -> 5.0.1

NickCao committed:
Merge pull request #265813 from aaronjheng/pb

pb: 0.1.0 -> 0.2.0

NickCao committed:
Merge pull request #265811 from fabaff/pysml-bump

python311Packages.pysml: 0.1.0 -> 0.1.1

NickCao committed:
Merge pull request #265810 from fabaff/tplink-omada-client-bump

python311Packages.tplink-omada-client: 1.3.5 -> 1.3.6

mweinelt committed:
Merge pull request #265773 from r-ryantm/auto-update/python310Packages.faster-whisper

python310Packages.faster-whisper: 0.8.0 -> 0.9.0

NickCao committed:
Merge pull request #265808 from fabaff/pycep-parser-bump

python311Packages.pycep-parser: 0.4.1 -> 0.4.2

NickCao committed:
Merge pull request #265807 from fabaff/pyoverkiz-bump

python311Packages.pyoverkiz: 1.12.1 -> 1.12.2

NickCao committed:
Merge pull request #265806 from fabaff/rns-bump

python311Packages.rns: 0.6.5 -> 0.6.6

NickCao committed:
Merge pull request #265805 from fabaff/plugwise-bump

python311Packages.plugwise: 0.34.4 -> 0.34.5

NickCao committed:
Merge pull request #265804 from fabaff/meshtastic-bump

python311Packages.meshtastic: 2.2.11 -> 2.2.12

NickCao committed:
Merge pull request #265802 from fabaff/publicsuffixlist-bump

python311Packages.publicsuffixlist: 0.10.0.20231104 -> 0.10.0.20231105

r-ryantm committed:
lsp-plugins: 1.2.12 -> 1.2.13

NickCao committed:
Merge pull request #265799 from fabaff/evohome-async-bump

python311Packages.evohome-async: 0.4.3 -> 0.4.4

NickCao committed:
Merge pull request #265798 from fabaff/blinkpy-bump

python311Packages.blinkpy: 0.22.2 -> 0.22.3

NickCao committed:
Merge pull request #265809 from fabaff/tldextract-bump

python311Packages.tldextract: 5.0.1 -> 5.1.0

NickCao committed:
Merge pull request #265788 from r-ryantm/auto-update/python310Packages.gocardless-pro

python310Packages.gocardless-pro: 1.47.0 -> 1.48.0

NickCao committed:
Merge pull request #265791 from fabaff/alexapy-bump

python311Packages.alexapy: 1.27.7 -> 1.27.8

NickCao committed:
Merge pull request #265819 from r-ryantm/auto-update/python310Packages.gptcache

python310Packages.gptcache: 0.1.41 -> 0.1.42

NickCao committed:
Merge pull request #265772 from r-ryantm/auto-update/python310Packages.ezyrb

python310Packages.ezyrb: 1.3.0.post2309 -> 1.3.0.post2311

NickCao committed:
Merge pull request #265774 from r-ryantm/auto-update/python310Packages.fasttext-predict

python310Packages.fasttext-predict: 0.9.2.1 -> 0.9.2.2

NickCao committed:
Merge pull request #265775 from r-ryantm/auto-update/python310Packages.finvizfinance

python310Packages.finvizfinance: 0.14.6 -> 0.14.7

NickCao committed:
Merge pull request #265778 from r-ryantm/auto-update/python310Packages.flask-paginate

python310Packages.flask-paginate: 2023.10.8 -> 2023.10.24

NickCao committed:
Merge pull request #265784 from r-ryantm/auto-update/python310Packages.glcontext

python310Packages.glcontext: 2.4.0 -> 2.5.0

NickCao committed:
Merge pull request #265786 from honnip/update-pretendard

pretendard: 1.3.8 -> 1.3.9

NickCao committed:
Merge pull request #265803 from fabaff/lsassy-bump

python311Packages.lsassy: 3.1.8 -> 3.1.9

NickCao committed:
Merge pull request #265763 from r-ryantm/auto-update/python310Packages.dvclive

python310Packages.dvclive: 3.1.0 -> 3.2.0

NickCao committed:
Merge pull request #265766 from r-ryantm/auto-update/python310Packages.easydict

python310Packages.easydict: 1.10 -> 1.11

wegank committed:
Merge pull request #265824 from elohmeier/mkpasswd-clang

mkpasswd: fix build with clang

NickCao committed:
Merge pull request #265767 from kashw2/terragrunt

terragrunt: 0.53.0 -> 0.53.2

wegank committed:
Merge pull request #265831 from klchen0112/librime

librime: set darwin as support platforms

wegank committed:
Merge pull request #264619 from mdarocha/nuget-arm

nuget: allow aarch64-linux and darwin builds

natsukium committed:
Merge pull request #265835 from GaetanLepage/jaxopt

python311Packages.jaxopt: 0.8.1 -> 0.8.2

mweinelt committed:
python311Packages.lru-dict: 1.2.0 -> 1.3.0

https://github.com/amitdev/lru-dict/releases/tag/v1.3.0

NickCao committed:
nvidia-vaapi-driver: 0.0.10 -> 0.0.11

Diff: https://github.com/elFarto/nvidia-vaapi-driver/compare/v0.0.10...v0.0.11

Changelog: https://github.com/elFarto/nvidia-vaapi-driver/releases/tag/v0.0.11

nikstur committed:
Merge pull request #261376 from secufoe/rshim-user-space-bfb-install

rshim-user-space: add bfb-install

r-ryantm committed:
python310Packages.jug: 2.3.0 -> 2.3.1

mweinelt committed:
home-assistant: backport litterrobot tests fix

The litterrobot tests broke when DST changed, so now they've pinned the
date and the tests should keep working.

pbsds committed:
Merge pull request #251858 from maxbrunet/feat/devpod

devpod: init at 0.4.1

matt-savvy committed:
Gleam: 0.31.0 -> 0.32.2

https://github.com/gleam-lang/gleam/blob/v0.32.0/CHANGELOG.md

Djabx committed:
google-cloud-sdk: 446.0.1 -> 452.0.1 (#264589)

fabaff committed:
Merge pull request #265850 from r-ryantm/auto-update/python310Packages.intellifire4py

python310Packages.intellifire4py: 3.1.29 -> 3.1.30

roberth committed:
Merge pull request #264674 from teutat3s/fix-dockertools-prefetch-example

doc: fix dockerTools nix-prefetch-docker example

RaitoBezarius committed:
Revert "rsync: fix regression with _FORTIFY_SOURCE=2" (#265876)

This reverts commit ad38853459b9e62bffe9af2f326dc5a2933a1666.

This change while correct causes a mass-rebuild and needs to be resubmitted for staging.

r-ryantm committed:
python310Packages.litellm: 0.1.738 -> 0.11.1

mweinelt committed:
Merge pull request #265858 from mweinelt/lru-dict-1.3.0

python311Packages.lru-dict: 1.2.0 -> 1.3.0

mweinelt committed:
piper-phonemize: 2023.9.27-2 -> 2023.11.6-1

https://github.com/rhasspy/piper-phonemize/compare/refs/tags/2023.9.27-2...2023.11.6-1

mweinelt committed:
piper-tts: 2023.9.27-1 -> 2023.11.6-1

https://github.com/rhasspy/piper/releases/tag/v2023.11.6-1

r-ryantm committed:
python310Packages.mandown: 1.5.0 -> 1.6.0

mweinelt committed:
Merge pull request #265881 from mweinelt/piper-2023.11.6-1

piper-tts, piper-phonemize: 2023.11.6-1

msfjarvis committed:
spicetify-cli: 2.25.2 -> 2.26.0

Diff: https://github.com/spicetify/spicetify-cli/compare/v2.25.2...v2.26.0

gepbird committed:
osu-lazer: 2023.815.0 -> 2023.1026.0

mweinelt committed:
Merge pull request #230380 from graham33/feature/zwave-js-server_module

zwave-js: module init, zwave-js-server: init at 1.33.0

fabaff committed:
python311Packages.roonapi: 0.1.4 -> 0.1.5

Diff: https://github.com/pavoni/pyroon/compare/0.1.4...0.1.5

Changelog: https://github.com/pavoni/pyroon/releases/tag/0.1.5

figsoda committed:
typos: 1.16.22 -> 1.16.23

Diff: https://github.com/crate-ci/typos/compare/v1.16.22...v1.16.23

Changelog: https://github.com/crate-ci/typos/blob/v1.16.23/CHANGELOG.md

fabaff committed:
python311Packages.twilio: 8.10.0 -> 8.10.1

Diff: https://github.com/twilio/twilio-python/compare/refs/tags/8.10.0...8.10.1

Changelog: https://github.com/twilio/twilio-python/blob/8.10.1/CHANGES.md

fabaff committed:
python311Packages.gios: 3.2.0 -> 3.2.1

Diff: https://github.com/bieniu/gios/compare/refs/tags/3.2.0...3.2.1

Changelog: https://github.com/bieniu/gios/releases/tag/3.2.1

fabaff committed:
python311Packages.nextdns: 2.0.0 -> 2.0.1

Diff: https://github.com/bieniu/nextdns/compare/refs/tags/2.0.0...2.0.1

Changelog: https://github.com/bieniu/nextdns/releases/tag/2.0.1

fabaff committed:
trivy: 0.46.1 -> 0.47.0

Diff: https://github.com/aquasecurity/trivy/compare/refs/tags/v0.46.1...v0.47.0

Changelog: https://github.com/aquasecurity/trivy/releases/tag/v0.47.0

figsoda committed:
git-mit: 5.12.169 -> 5.12.171

Diff: https://github.com/PurpleBooth/git-mit/compare/v5.12.169...v5.12.171

Changelog: https://github.com/PurpleBooth/git-mit/releases/tag/v5.12.171

JakeHPark committed:
lunarvim: init at 1.3.0

figsoda committed:
jaq: 1.1.0 -> 1.1.2

Diff: https://github.com/01mf02/jaq/compare/v1.1.0...v1.1.2

Changelog: https://github.com/01mf02/jaq/releases/tag/v1.1.2

figsoda committed:
xq-xml: 1.2.2 -> 1.2.3

Diff: https://github.com/sibprogrammer/xq/compare/v1.2.2...v1.2.3

Changelog: https://github.com/sibprogrammer/xq/releases/tag/v1.2.3

figsoda committed:
jql: 7.0.5 -> 7.0.6

Diff: https://github.com/yamafaktory/jql/compare/jql-v7.0.5...jql-v7.0.6

Changelog: https://github.com/yamafaktory/jql/releases/tag/jql-v7.0.6

r-ryantm committed:
python310Packages.mkdocs-git-revision-date-localized-plugin: 1.2.0 -> 1.2.1

r-ryantm committed:
python310Packages.mkdocs-jupyter: 0.24.5 -> 0.24.6

Artturin committed:
Merge pull request #264684 from ShamrockLee/shellcheck-minimal

risicle committed:
Merge pull request #265445 from giorgiga/gomplate-fix-build

gomplate: force build with go 1.20

thiagokokada committed:
Merge pull request #265909 from gepbird/osu-lazer-bump

osu-lazer: 2023.815.0 -> 2023.1026.0

pbsds committed:
Merge pull request #265547 from numinit/update-ovftool

ovftool: 4.5.0 -> 4.6.2 (4.6.0 on i686-linux)

veprbl committed:
Merge pull request #265514 from xworld21/treewide-texlive-with-packages

treewide: switch to texlive.withPackages

pbsds committed:
Merge pull request #248644 from r-ryantm/auto-update/aravis

aravis: 0.8.28 -> 0.8.30

pbsds committed:
Merge pull request #265548 from numinit/mattermost-8.1.4

mattermost: 8.1.3 -> 8.1.4

Mic92 committed:
Merge pull request #265906 from figsoda/typos

typos: 1.16.22 -> 1.16.23

ck3d committed:
Merge pull request #265656 from ck3d/vdr-updates-2023-11

vdrPlugins: updates 2023/11

symphorien committed:
Merge pull request #264985 from paveloom/gut

gut: 0.2.10 -> 0.3.0

GaetanLepage committed:
python311Packages.objax: 1.7.0 -> 1.8.0

Changelog: https://github.com/google/objax/releases/tag/v1.8.0

fabaff committed:
qovery-cli: 0.74.2 -> 0.74.3

Diff: https://github.com/Qovery/qovery-cli/compare/refs/tags/v0.74.2...v0.74.3

Changelog: https://github.com/Qovery/qovery-cli/releases/tag/v0.74.3

io12 committed:
try: fix homepage

fabaff committed:
cnspec: 9.5.1 -> 9.5.2

Diff: https://github.com/mondoohq/cnspec/compare/refs/tags/v9.5.1...v9.5.2

Changelog: https://github.com/mondoohq/cnspec/releases/tag/v9.5.2

fabaff committed:
Merge pull request #265905 from fabaff/trivy-bump

trivy: 0.46.1 -> 0.47.0

fabaff committed:
Merge pull request #265721 from fabaff/latexify-py-bump

python311Packages.latexify-py: 0.2.0 -> 0.3.1

fabaff committed:
Merge pull request #265904 from fabaff/roonapi-bump

python311Packages.roonapi: 0.1.4 -> 0.1.5

fabaff committed:
Merge pull request #265903 from fabaff/twilio-bump

python311Packages.twilio: 8.10.0 -> 8.10.1

fabaff committed:
Merge pull request #265902 from fabaff/gios-bump

python311Packages.gios: 3.2.0 -> 3.2.1

fabaff committed:
Merge pull request #265901 from fabaff/nextdns-bump

python311Packages.nextdns: 2.0.0 -> 2.0.1

fabaff committed:
Merge pull request #265815 from fabaff/skodaconnect-bump

python311Packages.skodaconnect: 1.3.7 -> 1.3.8

jbedo committed:
Merge pull request #263039 from NixOS/r-updates

rPackages: CRAN and BioC update

figsoda committed:
felix-fm: fix build on darwin

figsoda committed:
felix-fm: remove unused patch

pbsds committed:
Merge pull request #265801 from robert-manchester/gphoto2_fix_build

python311Packages.gphoto2: fix setuptools.__version__ build breakage

dotlambda committed:
Merge pull request #265269 from natsukium/restructuredtext-lint

python311Packages.restructuredtext-lint: rename from restructuredtext_lint

NickCao committed:
Merge pull request #265859 from NickCao/nvidia-vaapi

nvidia-vaapi-driver: 0.0.10 -> 0.0.11

wegank committed:
Merge pull request #265944 from io12/try-fix-homepage

try: fix homepage

sciyoshi committed:
nixos/lib/make-btrfs-fs: Use fakeroot and faketime (#265686)

wegank committed:
Merge pull request #265698 from Defelo/uiua

uiua: 0.0.25 -> 0.1.0

wegank committed:
Merge pull request #264609 from fpletz/pkgs/libdatachannel-0.19.2-includepath

libdatachannel: 0.19.2 -> 0.19.3, fix include path

wegank committed:
Merge pull request #264818 from mahmoudk1000/master

anki-bin: 2.1.66 -> 23.10

wegank committed:
Merge pull request #264411 from traxys/nominatim

nominatim: fix issues with package

NickCao committed:
Merge pull request #265929 from r-ryantm/auto-update/python310Packages.mkdocs-jupyter

python310Packages.mkdocs-jupyter: 0.24.5 -> 0.24.6

Artturin committed:
Merge pull request #265439 from ProminentRetail/autokey

NickCao committed:
Merge pull request #265923 from r-ryantm/auto-update/python310Packages.mkdocs-git-revision-date-localized-plugin

python310Packages.mkdocs-git-revision-date-localized-plugin: 1.2.0 -> 1.2.1

wegank committed:
Merge pull request #265202 from anthonyroussel/wiremock

wiremock: 2.35.0 -> 3.2.0

NickCao committed:
Merge pull request #265914 from figsoda/jql

jql: 7.0.5 -> 7.0.6

NickCao committed:
Merge pull request #265911 from figsoda/xq

xq-xml: 1.2.2 -> 1.2.3

NickCao committed:
Merge pull request #265910 from figsoda/jaq

jaq: 1.1.0 -> 1.1.2

NickCao committed:
Merge pull request #265907 from figsoda/mit

git-mit: 5.12.169 -> 5.12.171

wegank committed:
Merge pull request #265505 from kirillrdy/routedns

routedns: 0.1.20 -> 0.1.51

NickCao committed:
Merge pull request #265893 from msfjarvis/hs/spicetify-2.26.0

spicetify-cli: 2.25.2 -> 2.26.0

wegank committed:
Merge pull request #264940 from r-ryantm/auto-update/nats-server

nats-server: 2.10.2 -> 2.10.4

wegank committed:
Merge pull request #265473 from saikatdas0790/update-cargo-leptos

cargo-leptos: 0.2.0 -> 0.2.1

NickCao committed:
Merge pull request #265943 from fabaff/qovery-cli-bump

qovery-cli: 0.74.2 -> 0.74.3

NickCao committed:
Merge pull request #265950 from fabaff/cnspec-bump

cnspec: 9.5.1 -> 9.5.2

wegank committed:
Merge pull request #265290 from YMSTNT/update-lunatask

lunatask: 1.7.7 -> 1.7.8

wegank committed:
Merge pull request #265604 from imincik/qgis-ltr-3.28.12

qgis-ltr: 3.28.11 -> 3.28.12

wegank committed:
Merge pull request #265865 from r-ryantm/auto-update/python310Packages.jug

python310Packages.jug: 2.3.0 -> 2.3.1

wegank committed:
Merge pull request #265847 from r-ryantm/auto-update/python310Packages.ignite

python310Packages.ignite: 0.4.12 -> 0.4.13

wegank committed:
Merge pull request #265885 from r-ryantm/auto-update/python310Packages.mandown

python310Packages.mandown: 1.5.0 -> 1.6.0

NickCao committed:
Merge pull request #265851 from NickCao/sing-box

sing-box: 1.6.0 -> 1.6.1

wegank committed:
Merge pull request #265246 from MikaelFangel/update-ugrep

ugrep: 4.3.1 -> 4.3.2

wegank committed:
Merge pull request #265650 from sei40kr/etc-graphite-gtk-theme

graphite-gtk-theme: fix wallpapers

natsukium committed:
Merge pull request #265936 from GaetanLepage/objax

python311Packages.objax: 1.7.0 -> 1.8.0

wegank committed:
Merge pull request #265956 from figsoda/felix

felix-fm: fix build on darwin

wegank committed:
root: drop patches

wegank committed:
Merge pull request #265981 from wegank/root-patches-drop

root: drop patches

wegank committed:
Merge pull request #264876 from Rookeur/update-virtualbox-7-0-12

virtualbox: 7.0.10 -> 7.0.12

Commit:
addressing comments

Commit:
resolve merge conflict

Commit:
tidy up comment

cole-h commented:
I'm locking and closing this pull request because you accidentally requested reviews from a lot of people. Please open a new PR. To avoid this problem in the future, you can mark the PR as a draft before you rebase. While the PR is a draft, you will be able to see the list of maintainers that will be requested for review, before they all receive a notification.
